### ISSUE TITLE
Question about list_orm_many_to_one.html.twig

### ISSUE THREAD

**Issue by tim96** [sonata-project/SonataDoctrineORMAdminBundle#529]
Hello @OskarStark 
I try to show in list connected entities, like User and Cars for example. 
I add route in my admin class, but link to the connected entity will not show in list view(only text):

```php
->add('car', null, array('route' => array('name' => 'show')))
```

So I search why it happened, and I found this line of code in _list_orm_many_to_one.html.twig_:
`{% set route_role = route_name | upper %}`

```twig
{% if field_description.hasAssociationAdmin and field_description.associationadmin.hasRoute(route_name) and field_description.associationadmin.isGranted(route_role, value) %}
```

So I have strange situation my route has name 'Show' but I documentation we have only 'View' permission: https://sonata-project.org/bundles/admin/master/doc/reference/security.html#role-handler

if I write route name 'Show' -> I have route but without permissions.
If I write route name 'View' -> I don't have route but I have permissions.

Maybe you know how I can fix it?

**Comment by tim96** [comment]
The same problem discuss here: 
https://github.com/sonata-project/SonataAdminBundle/issues/1916#issuecomment-153093929
@greg0ire I saw you know a lot of about sonata admin bundle, maybe you can help?

**Comment by greg0ire** [comment]
Looks like this line is too naive :  `{% set route_role = route_name | upper %}`. What happens if you use `Admin::checkAccess` instead ?

**Comment by tim96** [comment]
@greg0ire Where do I need to use `Admin::checkAccess` ? In twig template?

**Comment by greg0ire** [comment]
Yes.

**Comment by greg0ire** [comment]
The current admin might be available in the `sonata_admin` variable.

**Comment by tim96** [comment]
What parameters I need to pass in this function?

**Comment by greg0ire** [comment]
The first argument should be `show`, and the second one `field_description.associationadmin`. Just look at the code inside `checkAccess`, it will probably make sense.

**Comment by tim96** [comment]
I add this line: `sonata_admin.checkAccess(route_name, field_description.associationadmin)`
And result: 
`Method "checkAccess" for object "Sonata\AdminBundle\Twig\GlobalVariables" does not exist in SonataDoctrineORMAdminBundle:CRUD:list_orm_many_to_one.html.twig at line 23`

**Comment by tim96** [comment]
This line: `admin.checkAccess(route_name, field_description.associationadmin)` return null.

**Comment by greg0ire** [comment]
It means `sonata_admin` is not what you are looking for. Figure out a way to get the current admin in the template.

**Comment by greg0ire** [comment]
Hmm… this method does not return anything, it just throws exceptions when there is no access…

**Comment by greg0ire** [comment]
Sadly it is the only one I know with the ('show' => 'VIEW') association.

**Comment by tim96** [comment]
I think the problem in that check operation `field_description.associationadmin.isGranted(route_role, value)`. and I am sure if I add _SHOW permission in security.yml this function return true.

**Comment by greg0ire** [comment]
I think the solution would be to introduce a method that would return 'VIEW' when given 'show'.

**Comment by tim96** [comment]
How we can reach it? 
I see only one solution to check this line of code ` {% set route_role = route_name | upper %}`

**Comment by greg0ire** [comment]
Can you rephrase? I don't understand.

**Comment by greg0ire** [comment]
I think `checkAccess` should be refactored so that it calls another public method, that would return a boolean, and throws the accessDeniedException when appropriate.

**Comment by tim96** [comment]
> I think the solution would be to introduce a method that would return 'VIEW' when given 'show'.

How we can do it? Can you explain your solution? 
At this moment I see only one solution: we need to change this line. Maybe exists another solution, but I need to dig code more deeper.

**Comment by greg0ire** [comment]
So `checkAccess` would call `hasAccess`, a method with the same arguments, but that returns true or false.

**Comment by greg0ire** [comment]
@Soullivaneuh what do you think?

**Comment by tim96** [comment]
@greg0ire Why we need this strange ('show' => 'VIEW') association? It is look like a small hack. Can we add view action and view route?

**Comment by greg0ire** [comment]
That's what I'm trying to find out, but no clue. @core23 @OskarStark  @rande , can you please help ?

**Comment by tim96** [comment]
@greg0ire Hi. I found in admin class this code:
```php
        $access = array_merge(array(
            'acl'                     => 'MASTER',
            'export'                  => 'EXPORT',
            'historyCompareRevisions' => 'EDIT',
            'historyViewRevision'     => 'EDIT',
            'history'                 => 'EDIT',
            'edit'                    => 'EDIT',
            'show'                    => 'VIEW',
            'create'                  => 'CREATE',
            'delete'                  => 'DELETE',
            'batchDelete'             => 'DELETE',
            'list'                    => 'LIST',
        ), $this->getAccessMapping());
```
And this check operation in twig:
```php
{% if admin.hasroute('history') and admin.id(object) and admin.isGranted('EDIT', object) %}
```
As you can see in some templates route name not equal access parameter name.
Maybe we can simple add something like this:
```php
{% if route_name|upper == 'SHOW %} {% set route_role = 'VIEW' %}
```
Or maybe create function to get access list for check access and call it in twig template.Using this function we can convert route name to route role:

```php
public function getAccessList()
{
        return (array(
            'acl'                     => 'MASTER',
            'export'                  => 'EXPORT',
            'historyCompareRevisions' => 'EDIT',
            'historyViewRevision'     => 'EDIT',
            'history'                 => 'EDIT',
            'edit'                    => 'EDIT',
            'show'                    => 'VIEW',
            'create'                  => 'CREATE',
            'delete'                  => 'DELETE',
            'batchDelete'             => 'DELETE',
            'list'                    => 'LIST',
        )
}
```
Or maybe create function like checkAccess but without exception?

**Comment by greg0ire** [comment]
> Or maybe create function like checkAccess but without exception?

This is what I proposed and would be the best solution IMO. Would you like to make a PR on the admin bundle ?

**Comment by tim96** [comment]
> This is what I proposed and would be the best solution IMO. Would you like to make a PR on the admin bundle ?
I don't like this solution for one reason. If we add new method it will be BC. If we have BC we need wait new version and it is can be a whole lot longer.

**Comment by greg0ire** [comment]
Adding a new method is BC indeed. And BC means : backwards compatible. So no need to wait for a new version.

**Comment by tim96** [comment]
If you add new method it will broken backward compatible.

**Comment by greg0ire** [comment]
> If you add new method it will broken backward compatible.

Of course not it won't! How would it?

**Comment by tim96** [comment]
For example you add in twig:
```php
{% if admin.hasAccess(route_name, object) %}
```
But version 2.3 for example don't have this method.

**Comment by tim96** [comment]
**hasAccess** it is new method that you want to add.

**Comment by greg0ire** [comment]
Oh ok you were speaking about the orm bundle

**Comment by tim96** [comment]
@greg0ire sure this code from orm bundle.

**Comment by greg0ire** [comment]
Adding something to the admin bundle is BC. Requiring something from a new version of the admin bundle means changing the `composer.json` version constraint for the admin bundle from the orm bundle to something like `2.3.11` if the new method is added then. And it can be considered a BC break if people can't upgrade to `2.3.11` of the admin bundle.

**Comment by tim96** [comment]
We can create two PR:
1. Manually convert route name to route role using rules from admin bundle.
2. Add new method for new release with brake BC
What do you think?

**Comment by greg0ire** [comment]
I think make 2 first, and if it is not merged quickly enough for you, make 1 .

**Comment by tim96** [comment]
@greg0ire ok, thank you. I will try to do it.

**Comment by core23** [comment]
Have you solved this @tim96 ?

**Comment by tim96** [comment]
> Have you solved this @tim96 ?

Not yet (((

**Comment by fracsi** [comment]
Any news on this? Just ran into this problem.
